### DISCUSSION TITLE
Question about multiple migration classes & order

### DISCUSSION THREAD

**Discussion by nazeem-soeltan** [flamingock/mongock#43] [Q&A] [answered]
Before the introduction of `ChangeUnit`, we had the following structure:
- A Package containing the 'main' ChangeLog (package net.company.changelogs) with order 001
- A package containing the ChangeLog with only index related changes (package net.company.changelogs.index) with order 002.

In certain database related tests where a db was spinned up, we only ran the i ChangeLog from package net.company.changelogs.index.

When the application was normally started up, we know that the main changelog ran first and then the one with the index changes due to the order that was specified as parameter in ChangeLog.

How can we achieve the same with ChangeUnits ? i.e make sure that units in a certain package are run first?  We still want to start the order with 001, 002 for classes that are in seperate packages.

**Comment by osantana85** [comment, marked as answer]
Hi @nazeem-soeltan , you can continue using the  “order” in ChangeUnit, the difference with legacy ChangeLogs is that you need to create a new ChangeUnit class for each new migration, instead of adding new ChangeSet method. So you will have a ChangeUnit class for each legacy ChangeSet, and each one will have its own order. Let me know if you need more details.

**Comment by nazeem-soeltan** [comment]
Just to confirm. We can have something like this:

`net.company.changelog` :
_V001...java
_V002..java

`net.company.indexlog`:
_V001...java
_V002...java


application will load classes from both packages.

In certain tests we only scan for `net.company.indexlog`.

Am I correct?

**Reply by dieppa** [reply]
Yes, in the production run, Mongock is supposed to load chanageUnits from both packages(assuming you are passing them). In test where you want to run only one of those packages, you need to let Mongock know by passing only that package.

You can even pass only one of those classes.

Keep in mind that the order should be unique among all the changeUnits, otherwise the order they are run will be probably unexpected and in the future Mongock will complain about it.

**Reply by nazeem-soeltan** [reply]
> Keep in mind that the order should be unique among all the changeUnits, otherwise the order they are run will be probably unexpected and in the future Mongock will complain about it.

This is exactly the thing we want to know. 
See image below on what we currently have:
[Image](https://imgur.com/0C1khTQ)

Altough its not a real problem, it feels a bit inconvenient to keep track of the order between all the packages. We'd hoped that we could have their own ordering per package.

**Reply by osantana85** [reply]
Currently it’s up to you, so you can prefix the order value with a number, i.e. “100001”, “100002”, “100003”… for the first package, and “200001”, “200002”, “200003”… for the second package. Hope this help you.
